lib/help_scout.rb

```diff
@@ -206,6 +208,8 @@ def request(method, path, options)
       raise NotFoundError
     when HTTP_INTERNAL_SERVER_ERROR
       raise InternalServerError
+    when HTTP_TOO_MANY_REQUESTS
+      raise TooManyRequestsError
```

Comment:
Would be nice if we can return the Rate Limiting headers: http://developer.helpscout.net/help-desk-api/#rate-limiting

Reply:
Interesting idea, what would you like to return? If we've hit the limit, I don't know what number makes sense there. The remaining amount of requests will probably be 0. 

We could add the amount of time to wait in the error message perhaps:

> A "Retry-After" header is returned indicating how many seconds to wait until retry. Applicable to all requests

Reply:
You can always check the headers directly via `client.last_response`, so if we just put in a 'nice' error message we at least give some feedback on the exception. Something a long the lines of "Rate limit (200 requests per minute) reached. You can try again after x seconds. See http://developer.helpscout.net/help-desk-api/#rate-limiting for more info". Or something.

Reply:
Pushed an update, let me know what you think. 

Kind of hard to test this IRL, but I think this covers it. Will keep an eye on it and update if it does turn out to be different in production.